iothub/device/src/InternalClient.cs

```diff
@@ -1054,11 +1001,16 @@ internal Task SendMethodResponseAsync(MethodResponseInternal methodResponse, Can
             }
         }
 
-        private Task EnableMethodAsync(CancellationToken cancellationToken)
+        private async Task HandleMethodEnableAsync(CancellationToken cancellationToken)
         {
-            return _deviceMethods == null && _deviceDefaultMethodCallback == null
-                ? InnerHandler.EnableMethodsAsync(cancellationToken)
-                : TaskHelpers.CompletedTask;
+            // If currently enabled, then skip
+            if (_isDeviceMethodEnabled)
```

Comment:
Now that `_deviceMethods` is no longer null, it can't be used as a determining factor. So we'll just rely on this boolean to control it.